src/bonjour.c

```diff
@@ -0,0 +1,102 @@
+#include <dns_sd.h>
+#include <stdio.h>
+#include "bonjour.h"
+#include "lxi.h"
+
+char *services[] = {
+    "_lxi._tcp",
+    "_vxi-11._tcp",
+};
```

Comment:
Here we just need to add "_scpi-raw._tcp", "_scpi-telnet._tcp", "_hislip._tcp".

Reply:
According to https://developer.apple.com/documentation/dnssd/1804742-dnsservicebrowse we can put these in one string so it will search for the services by calling DNSServiceBrowse() once.

Reply:
I've added the extra service types as a comma separated string, but the DNSServiceBrowse function returns a error code -65540 (from [this code](https://github.com/pieceofsummer/TouchRemote/blob/master/foo_touchremote/TouchRemote.Bonjour/DnsServiceErrorType.cs), it would indicate a bad param)

So I added a loop as per the Avahi code, but this seems to only broadcast the first string in the array because this is the only response back from the device. ie. all other types are ignored or not broadcast. Is this likely because the info pointer is not copied for each iteration?

``` c
void create_service_browser(char *reg_type, lxi_info_t *info) {
    DNSServiceRef service;
    DNSServiceErrorType error;

    error = DNSServiceBrowse(&service, 0, 0, reg_type, "local.", browse_callback, info);
    if(error != kDNSServiceErr_NoError) {
        fprintf(stderr, "DNSServiceBrowse() failed: %d\n", error);
        return;
    }

    error = DNSServiceProcessResult(service);
    if(error != kDNSServiceErr_NoError) {
        fprintf(stderr, "DNSServiceProcessResult() failed: %d\n", error);
        return;
    }

    DNSServiceRefDeallocate(service);
}

int bonjour_discover(lxi_info_t *info, int timeout) {
    DNSServiceRef service;
    DNSServiceErrorType error;
    int i;
    int service_count = sizeof(services) / sizeof(services[0]);

    // Loop over services
    for (i = 0; i < service_count; i++) {
        // create service browser
        create_service_browser(services[i], info);
    }
        return LXI_OK;
}
```